### DISCUSSION TITLE
Error handling in app router does not working [appDir]

### DISCUSSION THREAD

**Discussion by Andrew0914** [vercel/next.js#52065] [Help] [answered]
### Summary

I followed de conventiosn and create

* error.tsx
*  not-found.tsx
* global-error.tsx

But any of them catch my errors 500 or 404, default next js error paged still showing instead of my custom ones.

![image](https://github.com/vercel/next.js/assets/7088553/0afe0936-3a82-429a-b7b7-39894f919317)

### Additional information

```js
Uusing experimental appDir 
Nextjs 13.2.4

Implemented official example internationalisation under [lang] folder https://github.com/vercel/next.js/tree/canary/examples/app-dir-i18n-routing
```

**Comment by rahXephonz** [comment, marked as answer]
Hello

i think for custom error page you can use not-found page for one global 404 pages

or using a custom `app/[others]/page.tsx` directory and then in the page.tsx

```ts
import { redirect } from "next/navigation";

export default async function NotFound() {
    redirect("/404");
}
```

when you go to path like ``/test`` it will redirected to 404 pages 

instead of that theres another solutions for your question

you can use ``notFound()`` fn from next/navigation if theres checking a logic if you using middleware or API and then create ``/app/not-found.tsx`` pages

```tsx
//not-found.tsx
export default async function NotFound() {
    return <div>Not Found Pages</div>
}
```

theres reference issue related to your questions
https://nextjs.org/docs/app/api-reference/functions/not-found
Github Issue https://github.com/vercel/next.js/issues/45834

**Reply by Andrew0914** [reply]
Thanks a lot, your answer solved my problem ✅ , maybe is in progress but I think error.tsx or global-error.tsx should work as expected in app dir.

**Comment by RafalChojnowskiPagepro** [comment]
Why isn't global-error catching it? Shouldn't it be an ultimate boundary?
